### PR TITLE
Migrate to Realm

### LINKED ISSUE #44: Use a real database
Migrate clip storage to an actual, real database.

### PR DESCRIPTION
**Pull request by benashman**
👻

Fixes #44

### CODE REVIEW

File: Camera/EditClipViewController.swift [hunk 1]
```diff
@@ -45,33 +46,23 @@ class EditClipViewController: UIViewController, UITextFieldDelegate, UIGestureRe
...
             let image = UIImage(contentsOfFile: filePath)
             let imageView = UIImageView(image: image)
-            print(image)
             
             imageView.frame = self.view.bounds
-            imageView.backgroundColor = UIColor.blueColor()
             clipView.addSubview(imageView)
             
-        } else if fileExtension == ".mov" {
+        } else if clip.type == "video" {
```
Comment (andytlr) [inline Camera/EditClipViewController.swift:61]: Niiice. This was super lol.

File: Camera/ListViewViewController.swift [hunk 2]
```diff
@@ -27,8 +25,9 @@ class ListViewViewController: UIViewController, UITableViewDataSource, UITableVi
     }
     
     func updateTableView() {
-        clips = returnContentsOfTemporaryDocumentsDirectory()
-        clipCount = clips.count
+        let realm = try! Realm()
+        clips = realm.objects(Clip).sorted("filename", ascending: false)
```
Comment (andytlr) [inline Camera/ListViewViewController.swift:29]: I like that this is explicit now. And was thinking we could perhaps have the most recent clips at the top of the list but export and preview them all together in chronological order.

File: Camera/ListViewViewController.swift [hunk 3]
```diff
@@ -116,13 +108,15 @@ class ListViewViewController: UIViewController, UITableViewDataSource, UITableVi
         alertController.addAction(cancelAction)
         
         let destroyAction = UIAlertAction(title: "Yep, delete them.", style: .Destructive) { (action) in
+            let realm = try! Realm()
+            try! realm.write {
+                realm.deleteAll()
```
Comment (andytlr) [inline Camera/ListViewViewController.swift:115]: This doesn't work properly. Doing realm.deleteAll() is deleting the file references in the DB but we still need to delete the files or the app bundle will get huge.

File: Camera/PreviewViewController.swift [hunk 4]
```diff
@@ -25,31 +24,26 @@ class PreviewViewController: UIViewController {
...
         deleteLabel.alpha = 0
         keepLabel.alpha = 0
         
-        let latestItemInCameraRoll = String(cameraRoll.last!)
-        let latestFileName = cameraRoll.last!.lastPathComponent!
-        
-        let documentsDir = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)[0] as NSString
-        let filePath = documentsDir.stringByAppendingPathComponent("/tmp/\(latestFileName)")
-        
-        // Dis string bullshit to get file type
```
Comment (andytlr) [inline Camera/PreviewViewController.swift:48]: wukkkka

File: Camera/PreviewViewController.swift [hunk 5]
```diff
@@ -188,11 +182,12 @@ class PreviewViewController: UIViewController {
...
                         
-                        let cameraRoll = returnContentsOfTemporaryDocumentsDirectory()
-                        let latestFileName = cameraRoll.last!.lastPathComponent!
-                        removeItemFromDocumentsDirectory(latestFileName)
-                        
                         self.killPreviewAndRestartCamera()
+                        
+                        let realm = try! Realm()
+                        try! realm.write {
+                            realm.delete(self.clip)
```
Comment (andytlr) [inline Camera/PreviewViewController.swift:193]: This also doesn't work properly for the same reason.

**Comment by andytlr** [comment]
Awesome man. Only one issue I could see in two places. Deleting references in the DB still needs to also delete the file in the documents directory because it's persistent.

**Comment by benashman** [comment]
Code review FTW! Just fixed these two issues. Ship dis sucka?

**Pull request merged**